### ISSUE TITLE
sqs trigger lambda image issue

### ISSUE THREAD

**Issue by yangzhang21** [localstack/localstack#3890]
# Type of request: This is a ...

[X ] bug report
[ ] feature request

# Detailed description
Similar deployment/configuration as the issue described in https://github.com/localstack/localstack/issues/3871

1. I created a dummy lambda, which just print out a string and return
2. I built a docker image for this lambda and pushed it to localstack ecr repo, localhost:4510
3. I  created an iam role and attached it to a new lambda function
4. all above steps ran successfully
5. I use a sqs msg to invoke the lambda
6. I see multiple lambdas function instances invoked
7. after a while (after lambda timeout), I see error msg
```
2021-04-16T02:50:36:WARNING:bootstrap.py: Thread run method <function LambdaExecutor.execute.<locals>.do_execute at 0x7ffb595c0430>(None) failed: An error occurred (ReceiptHandleIsInvalid) when calling the DeleteMessageBatch operation: The input receipt handle is invalid.
``` 
...

## Expected behavior
invoke once and one lambda instance will be invoked
...

## Actual behavior
multiple lambda instances are invoked and see error msg
...

# Steps to reproduce
similar to https://github.com/localstack/localstack/issues/3871 
event-source-mapping cmd
```
aws lambda create-event-source-mapping --endpoint http://localhost:4566 \
                                       --function-name lsretriever \
                                       --batch-size 1 \
                                       --event-source-arn arn:aws:sqs:us-west-2:000000000000:test_queue
```
send msg cmd
```
aws sqs send-message --endpoint http://localhost:4566 \
                     --queue-url http://localhost:4566/000000000000/test_queue \
                     --message-body "This is a test"
```

## Command used to start LocalStack
 TMPDIR=/private$TMPDIR docker-compose up

...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by yangzhang21** [comment]
somehow I think I might find the possible reason for this scenario. it's because when i create the queue, i set visibility timeout too short. It should be longer than the function timeout. After I made the change, the multi-instance invocation issue seems solved. But then I found another issue. After the queue and lambda functions are deployed, if I continue to send messages to sqs to trigger more lambdas, I see they are invoked but not executed properly. I can see the lambda containers are created, but the lambda function is not executed. If I check logs, it only print the below information
```
$ docker logs ad38f4df0bf3
Using Lambda API Runtime target host: 'lsretriever.us-west-2.localhost.localstack.cloud:4566'
chown: invalid user: ‘sbx_user1051’
Starting daemons...
/tmp/ce6cbb53.sh: line 261: iptables: command not found
Host 'lsretriever.us-west-2.localhost.localstack.cloud' resolves to '172.17.0.2'
/tmp/ce6cbb53.sh: line 283: which: command not found
Starting XRay server loop on UDP port 2000
Starting DNS server loop on UDP port 53
```
after function times out, I get the error message
```
localstack_main | 2021-04-19T00:17:06:INFO:localstack.services.awslambda.lambda_executors: Running lambda cmd: docker run --user=root --cap-add=NET_ADMIN --entrypoint=/tmp/77ced61d.sh --dns 127.0.0.1 -e LOCALSTACK_DEBUG=1 -e AWS_LAMBDA_FUNCTION_MEMORY_SIZE=123 -e _LAMBDA_RUNTIME=None -e AWS_LAMBDA_LOG_GROUP_NAME="/aws/lambda/lsretriever" -e AWS_LAMBDA_LOG_STREAM_NAME=test -e AWS_LAMBDA_RUNTIME_API="lsretriever.us-west-2.localhost.localstack.cloud:4566" -e LOCALSTACK_HOSTS_ENTRY="lsretriever.us-west-2.localhost.localstack.cloud" -v /private/var/folders/jk/vg7nnbhj4qg0r0xz3yhskylw0000gn/T/:/tmp -d -e AWS_LAMBDA_FUNCTION_MEMORY_SIZE=123 -e _LAMBDA_RUNTIME=None -e AWS_LAMBDA_LOG_GROUP_NAME="/aws/lambda/lsretriever" -e AWS_LAMBDA_LOG_STREAM_NAME=test -e AWS_LAMBDA_RUNTIME_API="lsretriever.us-west-2.localhost.localstack.cloud:4566" -e LOCALSTACK_HOSTS_ENTRY="lsretriever.us-west-2.localhost.localstack.cloud" -e LOGGING_LEVEL="$LOGGING_LEVEL" -e LOCALSTACK_HOSTNAME="$LOCALSTACK_HOSTNAME" localhost:4510/lsretriever:latest "retriever_lambda.retriever_handler"
localstack_main | 2021-04-19T00:17:15:INFO:localstack_ext.services.awslambda.lambda_extended: Unable to invoke Lambda "arn:aws:lambda:us-west-2:000000000000:function:lsretriever": 
localstack_main | 2021-04-19T00:17:15:WARNING:bootstrap.py: Thread run method <function LambdaExecutor.execute.<locals>.do_execute at 0x7f10b932ae50>(None) failed:  Traceback (most recent call last):
localstack_main |   File "/opt/code/localstack/localstack/utils/bootstrap.py", line 682, in run
localstack_main |     result = self.func(self.params)
localstack_main |   File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 168, in do_execute
localstack_main |     return _run(func_arn=func_arn)
localstack_main |   File "/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py", line 149, in wrapped
localstack_main |     raise e
localstack_main |   File "/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py", line 145, in wrapped
localstack_main |     result = func(*args, **kwargs)
localstack_main |   File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 159, in _run
localstack_main |     raise e
localstack_main |   File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 147, in _run
localstack_main |     result = self._execute(func_arn, func_details, event, context, version)
localstack_main |   File "/opt/code/localstack/localstack/services/awslambda/lambda_executors.py", line 325, in _execute
localstack_main |     result = self.run_lambda_executor(cmd, stdin, env_vars=environment, func_details=func_details)
localstack_main |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/awslambda/lambda_extended.py", line 327, in run_lambda_executor
localstack_main |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/awslambda/lambda_extended.py", line 241, in run_lambda_executor
localstack_main |   File "/usr/lib/python3.8/queue.py", line 178, in get
localstack_main |     raise Empty
localstack_main | _queue.Empty
localstack_main | 
```

**Comment by whummer** [comment]
Thanks for reporting @yangzhang21 - this should be fixed in the meantime. Can you please give it another try with the `latest` version of the Docker image? Please keep us posted how it goes. Thanks!
